### PR TITLE
Add candidate search in internship. Change ordering.

### PR DESCRIPTION
**Pull request by vldmrklnk**

### CODE REVIEW

File: InternshipBE/BL/Services/CandidateService.cs [hunk 1]
```diff
@@ -115,5 +115,12 @@ private async Task<List<Candidate>> FilterCandidates(CandidateFilterModelDTO fil
 
             return await _candidates.ToListAsync();
         }
+
+        public async Task<List<CandidateDTO>> SearchByInternshipAsync(CandidateDTO searchModel)
```
Comment (romanstat) [inline InternshipBE/BL/Services/CandidateService.cs]: id

File: InternshipBE/BL/Services/CandidateService.cs [hunk 2]
```diff
@@ -115,5 +115,12 @@ private async Task<List<Candidate>> FilterCandidates(CandidateFilterModelDTO fil
...
+        {
+            var query = await _unitOfWork.Candidates.SearchCandidatesAsync(searchModel.Skip, searchModel.Take, searchModel.SearchText, searchModel.SortBy, searchModel.IsDesc, searchModel.InternshipId);
```
Comment (romanstat) [inline InternshipBE/BL/Services/CandidateService.cs]: candidates

File: InternshipBE/DAL/Interfaces/ICandidateRepository.cs [hunk 3]
```diff
@@ -16,6 +16,6 @@ public interface ICandidateRepository : IGenericRepository<Candidate>
 
         IQueryable<Candidate> GetCandidatesForFIlter();
 
-        Task<List<Candidate>> SearchCandidatesAsync(int skip, int take, string searchText, string sortBy, bool isDesc);
+        Task<List<Candidate>> SearchCandidatesAsync(int skip, int take, string searchText, string sortBy, bool isDesc, int InternshipId);
```
Comment (romanstat) [inline InternshipBE/DAL/Interfaces/ICandidateRepository.cs]: camelCase

File: InternshipBE/DAL/Repositories/CandidateRepository.cs [hunk 4]
```diff
@@ -52,9 +52,9 @@ public async Task<int> GetCandidatesCountAsync()
             return await _context.Candidates.CountAsync();
         }
 
-        public async Task<List<Candidate>> SearchCandidatesAsync(int skip, int take, string searchText, string sortBy, bool isDesc)
+        public async Task<List<Candidate>> SearchCandidatesAsync(int skip, int take, string searchText, string sortBy, bool isDesc, int InternshipId)
```
Comment (romanstat) [inline InternshipBE/DAL/Repositories/CandidateRepository.cs]: camelCase

**Review by romanstat — APPROVED** [review]

**Review by Evgeniy-Sinicin — APPROVED** [review]

**Review by Nikityanov — APPROVED** [review]

**Pull request merged**